app/src/main/java/com/keylesspalace/tusky/settings/AccountPreferenceHandler.kt

```diff
@@ -0,0 +1,47 @@
+package com.keylesspalace.tusky.settings
+
+import androidx.preference.PreferenceDataStore
+import com.keylesspalace.tusky.appstore.EventHub
+import com.keylesspalace.tusky.appstore.PreferenceChangedEvent
+import com.keylesspalace.tusky.db.AccountEntity
+import com.keylesspalace.tusky.db.AccountManager
+
+// TODO this must be possible with DI / @Inject somehow?
```

Comment:
The problem here is that this class a) is dependent on injected classes and thus must be injectable and injected itself (?) and b) depends on a configurable account instance.

Reply:
That means it always needs to work with the active account?

(Which would be somewhat hidden and tighter coupling than a handler that can work with any account.)

Reply:
You can't inject the active account (maybe it is possible with some assisted injection but I doubt that makes anything easier), so you either stick to your current solution, or apply my suggestion. I'm fine with both.

Reply:
Sticking (DI wishes removed).